src/dataflow/src/logging/materialized.rs

```diff
@@ -245,6 +294,26 @@ pub fn construct<A: Allocate>(
                 }
             });
 
+        let frontier_current = frontier.as_collection();
+
+        use differential_dataflow::operators::Count;
+        let kafka_consumer_info_current = kafka_consumer_info.as_collection().count().map({
+            let mut row_packer = repr::RowPacker::new();
+            move |((consumer_name, source_id, partition_id), pairs)| {
+                row_packer.pack(&[
+                    Datum::String(&consumer_name),
+                    Datum::String(&source_id.to_string()),
+                    Datum::String(&partition_id),
+                    Datum::Int64(pairs.element1.element1.element1),
+                    Datum::Int64(pairs.element1.element1.element2),
+                    Datum::Int64(pairs.element1.element2.element1),
+                    Datum::Int64(pairs.element1.element2.element2),
+                    Datum::Int64(pairs.element2.element1),
+                    Datum::Int64(pairs.element2.element2),
+                ])
+            }
```

Comment:
correspondingly:

```suggestion
            move |((consumer_name, source_id, partition_id), pairs)| {
                if let &[DiffPair {
                    element1: rxmsgs,
                    element2: rxbytes,
                }, DiffPair {
                    element1: txmsgs,
                    element2: txbytes,
                }, DiffPair {
                    element1: app_offset,
                    element2: consumer_lag,
                }] = &*pairs
                {
                    row_packer.pack(&[
                        Datum::String(&consumer_name),
                        Datum::String(&source_id.to_string()),
                        Datum::String(&partition_id),
                        Datum::Int64(rxmsgs),
                        Datum::Int64(rxbytes),
                        Datum::Int64(txmsgs),
                        Datum::Int64(txbytes),
                        Datum::Int64(app_offset),
                        Datum::Int64(consumer_lag),
                    ])
                } else {
                    unreachable!("bug: tried to unpack mismatching pairs.");
                }
            }
```

Reply:
there was a discussion a bit earlier about this, we're going with nested diff pairs so the compiler can optimize that into one 48 byte allocation instead of having a 24 byte vec struct + a 48 byte data buffer.

Reply:
oops, makes sense! The lispy way it is! (: